trunk/public/class-facebook-login-public.php

```diff
@@ -115,26 +115,65 @@ public function enqueue_scripts() {
 	 * @since   1.0.0
 	 */
 	public function print_button() {
-		$redirect = ! empty( $_GET['redirect_to'] ) ? esc_url($_GET['redirect_to']) : ( is_ssl() ? 'https://' : 'http://' ) . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+		$redirect = home_url() . '/' . $_SERVER['REQUEST_URI']; // @nicholas we discussed this, but still not sure this is sufficient/correct. Opinion?
```

Comment:
Actually, after reviewing the plugin code. For added security. Let have the ajax callback return the redirect_to url. Because right now the JS code is going to redirect to whatever is set in data-redirect which I don't like (https://github.com/Kluny/facebook-login/blob/master/trunk/public/js/facebook-login.js#L41). It wouldn't be hard for some script to trick the user and redirect it to somewhere else.

So return the actual redirect_to url from as part of the ajax response of this method: https://github.com/Kluny/facebook-login/blob/3f02f8c3f8337adc1b451ed430b71623224283b3/trunk/public/class-facebook-login-public.php#L346

This will make things more secure.

Reply:
I've changed this a bit. I found out that the redirect wasn't working at all. Eg if you're on https://fbelearning.ngrok.io/post-21/, it would redirect you to https://fbelearning.ngrok.io/. I moved this part:

``` $redirect = home_url() . '/' . $_SERVER['REQUEST_URI']; // @nicholas we discussed this, but still not sure this is sufficient/correct. Opinion?
-
-        if( ! empty( $_GET['redirect_to'] ) ) {
-               $server = strpos( $_GET['redirect_to'], home_url() );
-            if( $server === 0 ) {
-                $redirect = esc_url( $_GET['redirect_to'] );
-            }
-        }
```

to register_or_login_user(), but then it redirect users to /wp-admin/ajax.php. I changed it to use HTTP_REFERER, and now it works right. Referrer is still spoofable, so there's still a check for the site_url within it. Tests okay.